src/application/shell.ts

```diff
@@ -141,13 +147,27 @@ class ApplicationShell extends Widget {
   }
 
   /**
+   * A signal emitted when main area's active focus changes.
+   */
+  get activeChanged(): ISignal<any, FocusTracker.IChangedArgs<Widget>> {
```

Comment:
rather than handing out the focus tracker's signal, which exposes the focus tracker implementation detail as `sender`, you can define your own `activeChanged` signal and emit that signal from your `_onActiveChanged` method.